src/toast/Toast.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from "react";
+import { cx } from "@renderlesskit/react";
+
+import { useHover } from "../hooks";
+import { isFunction } from "../utils";
+import { Box, BoxProps } from "../box";
+import { forwardRefWithAs } from "../utils/types";
+import { useMergeRefs } from "../hooks/useMergeRefs";
+import { Toast as ToastType, useToasts } from "./RenderlessToast";
+import { useTheme } from "../theme";
+
+export type ToastContainerProps = BoxProps & {};
+
+export const ToastContainer = forwardRefWithAs<
+  ToastContainerProps,
+  HTMLDivElement,
+  "div"
+>((props, ref) => {
+  const { className, children, ...rest } = props;
+  const { toasts, updateHeight, calculateOffset } = useToasts();
+  const { hoverProps, isHovered } = useHover({});
+  const theme = useTheme();
+
+  const toastContainerStyles = cx(
+    theme.toast.container.base,
+    isHovered
+      ? theme.toast.container.hovered
+      : theme.toast.container.notHovered,
+    className,
+  );
+
+  return (
+    <Box ref={ref} className={toastContainerStyles} {...hoverProps} {...rest}>
+      {toasts.map((toast: ToastType, index) => {
+        const toastsLength = toasts.length;
+        const sortedIndex = toastsLength - (index + 1);
+
+        return (
+          <ToastWrapper
+            key={toast.id}
+            sortedIndex={sortedIndex}
+            toast={toast}
+            updateHeight={updateHeight}
+            isHovered={isHovered}
+            hoverOffset={calculateOffset(toast.id)}
+          />
+        );
+      })}
+    </Box>
+  );
+});
+
+ToastContainer.displayName = "ToastContainer";
+
+export type ToastWrapperProps = BoxProps & {
+  toast: ToastType;
+  sortedIndex: number;
+  isHovered: boolean;
+  hoverOffset: number;
+  updateHeight: (toastId: string, height: number) => void;
+};
+
+export const ToastWrapper = forwardRefWithAs<
+  ToastWrapperProps,
+  HTMLDivElement,
+  "div"
+>((props, ref) => {
+  const { toast, sortedIndex, updateHeight, isHovered, hoverOffset } = props;
+
+  const htmlRef = React.useCallback(
+    (el: HTMLElement | null) => {
+      if (el) {
+        const boundingRect = el.getBoundingClientRect();
+        toast.visible && updateHeight(toast.id, boundingRect.height);
+      }
+    },
+    [updateHeight, toast.id, toast.visible],
+  );
+
+  const clampedIndex = sortedIndex > 4 ? 4 : sortedIndex;
+  const translateYGap = -10 * clampedIndex;
+  const scalePercent = 1 - 0.05 * clampedIndex;
+  const showAlertContent = sortedIndex === 0 || isHovered;
```

Comment:
Maybe we can rename this variable to be more explicit and generic? 

`isToastShown`, `isContentShown` ?? 🤔